index carbon for IBM.com libraries

francinelucca opened the pull request:
Here's first 5 as promised @mattrosno @kennylam

Notes
- Didn't add any demoLinks to `ibmdotcom-styles` library as there is no generic storybook, correct? @mattrosno 
- Haven't added thumbnails yet so made a note to verify thumbnailPaths until I get them from design
- https://ibmdotcom-utilities.mybluemix.net doesn't seem to be working, is there another url? @kennylam

mattrosno commented on packages/react/carbon.yml:
```diff
@@ -0,0 +1,61 @@
+# yaml-language-server: $schema=https://unpkg.com/@carbon-platform/schemas/carbon-resources.schema.json
```
```suggestion
# yaml-language-server: $schema=https://unpkg.com/@carbon-platform/schemas@v1/carbon-resources.schema.json
```

Let's pin to v1.

mattrosno commented on packages/services/carbon.yml:
```diff
@@ -0,0 +1,77 @@
+# yaml-language-server: $schema=https://unpkg.com/@carbon-platform/schemas/carbon-resources.schema.json
```
```suggestion
# yaml-language-server: $schema=https://unpkg.com/@carbon-platform/schemas@v1/carbon-resources.schema.json
```

mattrosno commented on packages/react/carbon.yml:
```diff
@@ -0,0 +1,61 @@
+  id: ibmdotcom-react
+  name: IBM.com React
+  description: IBM.com React components and patterns
+  noIndex: true
```
```suggestion
```

We want the React components in the global index.

mattrosno commented on packages/utilities/carbon.yml:
```diff
@@ -0,0 +1,83 @@
+  id: ibmdotcom-utilities
+  name: IBM.com Utilities
+  description: IBM.com ES6 utility classes
+  externalDocsUrl: https://ibmdotcom-utilities.mybluemix.net
```
@kennylam is there a different documentation site than https://ibmdotcom-utilities.mybluemix.net?

mattrosno replied:
Should we just use this here too? https://carbon-design-system.github.io/carbon-for-ibm-dotcom/utilities

mattrosno commented on packages/react/carbon.yml:
```diff
@@ -0,0 +1,61 @@
+    - type: storybook
+      name: Storybook
+      action: link
+      url: https://www.ibm.com/standards/carbon/react/?path=/story/overview-getting-started--page
```
```suggestion
      url: https://www.ibm.com/standards/carbon/react
```

We can shorten this URL to ensure nothing breaks if they update their first page.

mattrosno commented on packages/services/carbon.yml:
```diff
@@ -0,0 +1,77 @@
+assets:
+  analytics-api:
+    name: AnalyticsAPI
+    description: Analytics API class with methods for firing analytics events on ibm.com.
```
```suggestion
    description: Analytics API class with methods for firing analytics events on IBM.com.
```

I know we're just copy/pasting existing descriptions, but we should capitalize ibm.com everywhere we display it on our website unless it's a URL. Maybe a find-all and capitalize after indexing all?

mattrosno commented on packages/utilities/carbon.yml:
```diff
@@ -0,0 +1,83 @@
+  externalDocsUrl: https://ibmdotcom-utilities.mybluemix.net
+  noIndex: true
+  demoLinks:
+    - type: storybook
+      name: Storybook
```
This library too, use `jsdoc` and `JSDoc`.

francinelucca replied:
going for externalDocsUrl instead

francinelucca replied:
using externalDocsUrl isntead

mattrosno commented on packages/web-components/carbon.yml:
```diff
@@ -0,0 +1,60 @@
+# yaml-language-server: $schema=https://unpkg.com/@carbon-platform/schemas/carbon-resources.schema.json
+library:
+  id: ibmdotcom-react
+  name: IBM.com React
+  description: IBM.com React components and patterns
+  externalDocsUrl: www.ibm.com/standards/web/carbon-for-ibm-dotcom
+  inherits: ibmdotcom-styles
+  demoLinks:
+    - type: storybook
+      name: Storybook
+      action: link
+      url: https://www.ibm.com/standards/carbon/react/?path=/story/overview-getting-started--page
```
This should be web components.

francinelucca replied:
Done!

mattrosno reviewed (commented):
>Didn't add any demoLinks to ibmdotcom-styles library as there is no generic storybook, correct?

Good call. I agree

---

We have function tags that you can apply for services and utilities. I know we won't immediately see these in the catalogs because we're applying `noIndex`s, but still good to do: https://github.com/carbon-design-system/carbon-platform/blob/main/docs/resource-schemas.md#function-tags

---

We'll want to add component tags to the styles package. Tags are inheritable so they only need set in the styles package. I'd ask @oliviaflory for help with these, and also @aubrey-oneal who helped define the tags. https://github.com/carbon-design-system/carbon-platform/blob/main/docs/resource-schemas.md#component-tags

---

Do we need to add `framework: vanilla` to the service assets?

mattrosno commented on packages/services/carbon.yml:
```diff
@@ -0,0 +1,77 @@
+    noIndex: true
+    externalDocsUrl: https://carbon-design-system.github.io/carbon-for-ibm-dotcom/services/AnalyticsAPI.html
+    demoLinks:
+      - type: storybook
+        name: Storybook
```
On it. https://github.com/carbon-design-system/carbon-platform/issues/393

francinelucca replied:
using externalDocsUrl instead

kennylam commented:
> https://ibmdotcom-utilities.mybluemix.net/ doesn't seem to be working, is there another url?

@francinelucca All the `mybluemix` environments are gone now, in anticipation of its sunsetting. The new URLs are:
- `utilities`: https://carbon-design-system.github.io/carbon-for-ibm-dotcom/utilities/
- `services`: https://carbon-design-system.github.io/carbon-for-ibm-dotcom/services/

Sorry about that! We need to get some of those URLs updated.

francinelucca commented:
> > Haven't added thumbnails yet so made a note to verify thumbnailPaths until I get them from design
> 
> @francinelucca The thumbnails are all here for when you're ready to add them: https://github.com/carbon-design-system/carbon-for-ibm-dotcom-website/tree/main/src/components/ComponentGallery/images

Thanks! @oliviaflory we had some design folks work on some enhanced thumbnails that are optimized for our use-case in the app; we'll go with those first and will let you know should we need anything else 🎉

francinelucca commented:
@mattrosno can I get a 2nd review?
- Added thumbnails
- Removed demoLinks from services & utilities
- Added vanilla framework to services
- Added tags (will schedule time with Aubrey and Olivia once I get all components in to double check on these classifications)

francinelucca commented:
This is ready for a preliminary review, finished indexing all 5 libraries.

Outstanding items:
- [ ]  Still need a couple thumbnails, will add to PR once I get files from @jeanservaas 
   Missing SVGs:
     - Card 
     - CTA Block 
     - Feature Section 
     - Language Selector 
     - Link List Section
     - Tag Link
- [ ]  Sent over current dotcom components tags categorization to @oliviaflory and @aubrey-oneal. Got approval from Aubrey and awaiting Olivia's feedback
- [ ] Need confirmation from @kennylam on doubt regarding duplicate services/utilities entries in storybooks
- [ ] @dotcom team:  is there a **react** storybook entry for `Lead Space Search`
- [ ] @mattrosno FYI, under @kennylam's guidance skipped indexing the following components for the **react** library since they're technically not react implementations but instead wrappers around the web-components, would like to hear your input on this.
   Skipped React Lib Comps:
   - Back to Top
   - Card in card
   - Card section carousel
   - Carousel
   - Content Group Banner
   - Tabs Extende
   - Tag Link
   - Leaving IBM Modal
   - Filter Panel
   - Tabs Extended Media

mattrosno commented:
@francinelucca @kennylam the end user doesn't care if a React component wraps a web component or not? All they know is that they are building with React, and any React component that's exported in the React package should be documented (indexed). I would go ahead and index those too.

https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/packages/react/src/index.js

mattrosno commented on packages/styles/carbon.yml:
```diff
@@ -0,0 +1,681 @@
+    type: component
+    platform: web
+    noIndex: true
+    thumbnailPath: ../../thumbnails/back-to-top.svg
```
Makes sense to put all thumbnails in this one directory for starters. We might want to move them into `/packages/styles` asset subdirectories if each asset has a `/docs` directory in the future for usage, design, and code docs. But until we get there, I like this simple approach.

mattrosno reviewed (commented):
Looking good! Let's chat on how we can test this locally before merging this PR. One thought, you could try:

```javascript
const devLibraries = {
  'ibmdotcom-react': {
    host: 'github.com',
    org: 'carbon-design-system',
    repo: 'carbon-for-ibm-dotcom',
    ref: 'francinelucca:feat/carbon-platform/index-libraries',
    path: '/packages/react',
    sponsor: 'ibm-dotcom',
    group: ibmdotcomGroup
  },
}
```

Which I **think** will index this branch locally for you instead of the default branch.

Also, it'd be cool to add a **secret** `&index=all` param or something to our catalogs so we could see everything to test.

francinelucca commented:
@mattrosno I'm good to go with this, you can test all these libs over on carbon-platform's `flucca/feat/index-dotcom` , set `INDEX_ALL=1` in .env.local to override `noIndex=true`

mattrosno commented on packages/utilities/carbon.yml:
```diff
@@ -0,0 +1,194 @@
+      - utility
+  escape-reg-exp:
+    name: escapeRegExp
+    description: Utiltity function for escaping regex expressions https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#Escaping.
```
Remove Mozilla link?

francinelucca replied:
Done!

mattrosno commented on packages/services/carbon.yml:
```diff
@@ -0,0 +1,116 @@
+      - service
+  ddo-api:
+    name: DDOAPI
+    description: DDO API class with methods of fetching search results for ibm.com.
```
Can you do a find-all for ` ibm.com` to update to ` IBM.com` in descriptions? Since these descriptions show up in the catalog list view, we don't want to have lowercase `ibm.com` when not in a URL.

francinelucca replied:
Done!

mattrosno reviewed (commented):
I previewed these 5 libraries in Carbon platform and everything looks good on my end. As I'm not on this squad I defer approving to yall; the format of this information looks good to me.

kennylam reviewed (approved):
LGTM thanks @francinelucca!
